resource_type on FileSetPresenter

hectorcorrea opened the pull request:
...and therefore the test needs to updated.

jcoyne commented:
I was wondering if ScholarlyArticle and AudioObject aren't subclasses of CreativeWork.  If that's the case, we ought to be using the more specific terms.

jcoyne commented:
@hectorcorrea here's a hint: https://github.com/projecthydra-labs/curation_concerns/blob/master/curation_concerns-models/lib/generators/curation_concerns/models/templates/config/curation_concerns.rb#L34

mjgiarlo commented:
:+1: to using the more specific terms.

hectorcorrea commented:
Thanks for the pointers. I've updated the code to pick the correct `resource_type` which is why it wasn't working. `resource_type` must have been a delegate in a previous version of the code. It is not anymore and therefore the code was throwing an exception. I updated the code to pick the `resource_type` from the `solr_document` and now the execution continues and the tests pass.

jcoyne commented:
Should FileSet's have a resource_type? Or should this just be for works?

mjgiarlo commented:
@jcoyne I don't think FileSets should have a resource_type, no; just works.

hectorcorrea commented:
@mjgiarlo @jcoyne What does that mean for these two tests? https://github.com/projecthydra-labs/sufia-core/blob/master/spec/presenters/sufia/file_set_presenter_spec.rb#L37-L53

Are the tests not valid anymore because they are for the FileSetPresenter? Should these tests be moved to a WorkPresenterSpec ?

mjgiarlo commented:
You've got it, @hectorcorrea!

hectorcorrea commented:
@jcoyne @mjgiarlo I think this is what you two wanted, but I am not sure. Let me know what you think. 

Notice that the `work presenter` in CurationConcerns is named `work_show_presenter` (rather than `generic_work_presenter`)

mjgiarlo commented on app/presenters/sufia/work_show_presenter.rb:
```diff
@@ -12,7 +12,8 @@ def tweeter
     def itemtype
       # Look up the first non-empty resource type value in a hash from the config
-      Sufia.config.resource_types_to_schema[resource_type.to_a.reject(&:empty?).first] || 'http://schema.org/CreativeWork'
+      resource_type = solr_document.resource_type.to_a.reject(&:empty?).first
```
This is an improvement. :+1:

mjgiarlo commented:
Looks right to me! @jcoyne?

jcoyne commented on app/presenters/sufia/work_show_presenter.rb:
```diff
@@ -1,5 +1,5 @@
 module Sufia
-  class GenericWorkPresenter < ::CurationConcerns::GenericWorkPresenter
+  class WorkShowPresenter < ::CurationConcerns::WorkShowPresenter
```
I think we want this to subclass CurationConcerns::GenericWorkPresenter.  That's how people will customize the show page.

mjgiarlo replied:
Where is `CurationConcerns::GenericWorkPresenter` defined, @jcoyne? Or is it generated? I'm not seeing it in either CC or sufia-core.

hectorcorrea replied:
@jcoyne I didn't see `CurationConcerns::GenericWorkPresenter` either, but found `CurationConcerns::WorkShowPresenter` which is why I renamed the one in sufia-core.

jcoyne replied:
It's generated into the test application.

mjgiarlo replied:
I don't see it in my test application.

jcoyne replied:
Oh, yes, I was thinking Form.  I have no idea where that is/was from.

jcoyne replied:
Is it possible that this class was dead code prior to these changes?

jcoyne replied:
I think it might still be dead code. I don't see `WorkShowPresenter` referenced anywhere.

mjgiarlo replied:
Buh? `WorkShowPresenter` is referenced and used in CC. sufia-core references and uses `GenericWorkPresenter` in the `GenericWorksControllerBehavior`.

hectorcorrea commented:
Rebased. This takes care of the two tests that check for `resource_type` in rspec ./spec/presenters/sufia/file_set_presenter_spec.rb

I'll submit a separate PR for the issue in which the `generic_works_controller_behavior.rb` references the wrong class in `show_presenter` as well as a test for that condition.
